src/core/thread/tmf.hpp

```diff
@@ -82,8 +83,10 @@ class TmfAgent : public Coap::Coap
     bool IsTmfMessage(const Ip6::MessageInfo &aMessageInfo) const;
 
 private:
-    otError        Start(uint16_t aPort);
+    otError        Start(uint16_t aPort, otNetifIdentifier aNetifIdentifier);
     static otError Filter(const ot::Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo, void *aContext);
+
+    otNetifIdentifier mNetifIdentifier;
```

Comment:
Yes it can. I don't have strong opinion on this and there is no significant difference. 

If it's a parameter of `Start()`, we need to change `GetInstance().GetBackboneTmfAgent().Start()` to `GetInstance().GetBackboneTmfAgent().Start(OT_NETIF_BACKBONE)`. It seems a little redundant because `GetInstance().GetBackboneTmfAgent().Start(OT_NETIF_THREAD)` is illegal.

Reply:
I think we should have separate class for BackboneTmfAgent, which not only has a different port and netif, but also different filter rules. Thoughts?

Reply:
Note that the default port for Backbone TMF is 61631, same as the Thread TMF port. So, by default they share the same port number. 

I haven't investigated the filter rules. If they differ for Thread TMF and Backbone TMF, I would agree that it's better to have separate classes. 
But I think it's not the major issue. I will investigate and revise accordingly.

Reply:
I guess the backbone TMF should only allow link local scope destinations?

Reply:
this is resolved since there is now a separate class `BackboneTmfAgent` with customized TMF filter rules.